src/lib/draw/config.ts

```diff
@@ -0,0 +1,83 @@
+import type { FeatureUnion } from "types";
+import {
+  newFeatureId,
+  getArbitrarySchemeRef,
+  gjSchemeCollection,
+} from "lib/draw/stores";
+import type { FeatureWithAnyProps } from "lib/draw/types";
+import type { Feature, LineString, Polygon, Point } from "geojson";
+import { interventionName } from "lib/sidebar/scheme_data";
+import { schema } from "stores";
+import { get } from "svelte/store";
+
+// The draw code should be agnostic to the feature properties that differ by
+// schema. Start centralizing the logic here, so it's easy for other users to
+// override.
+
+// TODO As an alternate idea, users could implement a custom Svelte store with methods for doing these things
+
+export let cfg = {
+  interventionName: (f: FeatureWithAnyProps) => {
+    return interventionName(f as FeatureUnion);
+  },
+
+  newPointFeature: (f: Feature<Point>) => {
```

Comment:
The theme here is that these functions get a "mostly just geometry" object and then make it obey the invariants imposed by `gjSchemeCollection`. Some of the logic is common,but stuff like `intervention_type` and fiddling with route names is not.